### PR TITLE
198 search

### PR DESCRIPTION
**Pull request by NazarPo**
Implements #198

### CODE REVIEW

File: api/search/index.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import { Router } from "express";
+import { loadLeads } from "./searchAggregation";
+const router = new Router();
+
+// @route   GET api/searchResults
+// @desc    Search result by domain and regex
+// @access  Private
+
```
Comment (valentyntu) [inline api/search/index.js]: We don't make empty line between route and its description

File: api/search/index.test.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  createLead,
+  createStage,
+  createUserAndDomain,
+  dropTables,
+} from "../../test/db-prepare";
+
+const app = () => express(routes);
+const matchingContactName = "Matching contact";
+const defaultContactName = "Test contact";
+const leadType = "Lead";
```
Comment (valentyntu) [inline api/search/index.test.js]: Name constants like "LEAD_TYPE" to be sure what it is

File: client/src/modules/layouts/Navbar/Search/Search.tsx [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+          onFocus={this.onFocus}
+          onSelect={this.onSelect}
+          value={state.value}
+        />
+      </div>
+    );
+  }
+
+  private onSelect = (value: string, item: SearchItemModel) => {
+    window.location.replace(`../lead/${item._id}`);
```
Comment (valentyntu) [inline client/src/modules/layouts/Navbar/Search/Search.tsx]: forces page to reload, you shouldn't do this in single page app

File: client/src/modules/layouts/Navbar/searchActions.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
+import axios from 'axios';
+import { Dispatch } from 'redux';
+import { GET_ERRORS } from '../../../actionTypes';
+import { LOAD_SEARCH_RESULT, START_SEARCH_LOADING } from './types';
+
+export const loadSearchResult = (query:string) => (dispatch:Dispatch) => {
+  dispatch({ type: START_SEARCH_LOADING });
+  // setTimeout(2000);
```
Comment (valentyntu) [inline client/src/modules/layouts/Navbar/searchActions.ts]: remove unused code

File: client/src/modules/layouts/Navbar/searchReducer.ts [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+  loading: false,
+  result: [],
+};
+
+interface Action {
+  type: string,
+  payload: any;
+}
+
+
```
Comment (valentyntu) [inline client/src/modules/layouts/Navbar/searchReducer.ts]: remove redundant empty line

File: client/src/modules/layouts/Navbar/searchReducer.ts [hunk 6]
```diff
@@ -0,0 +1,23 @@
+import { LOAD_SEARCH_RESULT, START_SEARCH_LOADING } from './types';
+
+const initialState = {
+  loading: false,
+  result: [],
+};
+
+interface Action {
```
Comment (alexanderko) [inline client/src/modules/layouts/Navbar/searchReducer.ts]: use interface from Redux

**Pull request merged**